internal/tiltfile/k8s.go

```diff
@@ -887,3 +888,57 @@ func newK8sObjectID(e k8s.K8sEntity) k8sObjectID {
 func (s *tiltfileState) k8sContext(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
 	return starlark.String(s.kubeContext), nil
 }
+
+func (s *tiltfileState) allowK8SContexts(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var contexts starlark.Value
+	if err := starlark.UnpackArgs(fn.Name(), args, kwargs,
+		"contexts", &contexts,
+	); err != nil {
+		return nil, err
+	}
+
+	for _, c := range starlarkValueOrSequenceToSlice(contexts) {
+		switch val := c.(type) {
+		case starlark.String:
+			s.whitelistedK8SContexts = append(s.whitelistedK8SContexts, k8s.KubeContext(val))
+		default:
+			return nil, fmt.Errorf("allow_k8s_contexts contexts must be a string or a sequence of strings; found a %T", val)
+
+		}
+	}
+
+	s.whitelistedK8SContexts = append(s.whitelistedK8SContexts)
+
+	return starlark.None, nil
+}
+
+func (s *tiltfileState) validateK8SContext(kubeConfig *api.Config) error {
+	contextName := kubeConfig.CurrentContext
+	clusterName := kubeConfig.Contexts[contextName].Cluster
+	server := kubeConfig.Clusters[clusterName].Server
+	url, err := url.Parse(server)
+	if err != nil {
+		return errors.Wrapf(err, "running in kube context %q, which specifies kube api url %q. error parsing that url", contextName, server)
+	}
+
+	ipaddr, err := net.ResolveIPAddr("ip", url.Hostname())
```

Comment:
it seems very odd for Tiltfile execution to be making DNS requests. can we just whitelist a few hard-coded hosts (127.0.0.1, localhost)?

Reply:
Hm, it feels sloppy to me to poorly reimplement loopback detection, and I think I'd rather move the dns request outside of Tiltfile execution, but it's probably not worth the effort given some hard-coding is probably 99% good enough, so ok let's do that and not worry about it unless someone notices.

(My main hangup here is that at a former employer, dev machines were configured such that 127.0.0.1 resolved to `$USER.local`, so tools would detect that and talk to `$USER.local` instead of `localhost`. I have no idea why they did that or if that's common elsewhere, and I'm probably overindexing on that experience).

Reply:
fwiw, we're not trying to cover the case of "urls that J. Random Engineer has added manually", we're covering the case of "urls that Kubernetes tools will add to kubeconfig", which is probably much more limited.